src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/FileBinding.java

```diff
@@ -48,53 +43,11 @@
         return FileCredentials.class;
     }
 
-    @Override public SingleEnvironment bindSingle(Run<?,?> build, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
-        FileCredentials credentials = getCredentials(build);
-        FilePath secrets = secretsDir(workspace);
-        String dirName = UUID.randomUUID().toString();
-        final FilePath dir = secrets.child(dirName);
-        dir.mkdirs();
-        secrets.chmod(/*0700*/448);
+    @Override protected final FilePath write(FileCredentials credentials, FilePath dir) throws IOException, InterruptedException {
         FilePath secret = dir.child(credentials.getFileName());
-        copy(secret, credentials);
-        if (secret.isDirectory()) { /* ZipFileBinding */
-            // needs to be writable so we can delete its contents
-            // needs to be executable so we can list the contents
-            secret.chmod(0700);
-        }
-        else {
-            secret.chmod(0400);
-        }
-        return new SingleEnvironment(secret.getRemote(), new UnbinderImpl(dirName));
-    }
-    
-    private static class UnbinderImpl implements Unbinder {
```

Comment:
Cannot just delete this, as it could be present in running Pipeline builds. Can be deprecated, and optionally `readResolve` to the new version instead of implementing `unbind`.